sdk/core/Azure.Core/src/ResponseInnerError.cs

```diff
@@ -28,14 +30,29 @@ internal ResponseInnerError(string? code, string? message, ResponseInnerError? i
         public string? Code { get; }
 
         /// <summary>
-        /// Gets the error message.
+        /// Gets the inner error.
         /// </summary>
-        public string? Message { get; }
+        public ResponseInnerError? InnerError { get; }
 
         /// <summary>
-        /// Gets the inner error.
+        /// Attempts to retrieve a service specific property from the error.
         /// </summary>
-        public ResponseInnerError? InnerError { get; }
+        /// <param name="name">The name of the error.</param>
+        /// <param name="value">The variable to assign the value to.</param>
+        /// <typeparam name="T">The type of the requested property.</typeparam>
+        /// <returns><c>true</c> if the property exists, <c>false</c> otherwise.</returns>
+        public bool TryGetCustomProperty<T>(string name, out T? value)
```

Comment:
Is this only here to be used in tests?

Reply:
I had it as part of public implementation before we decided to move most things internal. Will remove.